executor/executor.go

```diff
@@ -1522,11 +1554,18 @@ func (e *UnionExec) Close() error {
 // Before every execution, we must clear statement context.
 func ResetContextOfStmt(ctx sessionctx.Context, s ast.StmtNode) (err error) {
```

Comment:
I suggest adding another de-attaching at [`recordSet.Close`](https://github.com/pingcap/tidb/blob/master/executor/adapter.go#L149), which is called after every execution.

Reply:
Almost LGTM, please address this comment @Yisaer .

Reply:
updated